src/tokens/tokens.service.ts

```diff
@@ -0,0 +1,68 @@
+import { HttpService } from '@nestjs/axios'
+import { Cache, CACHE_MANAGER } from '@nestjs/cache-manager'
+import { Injectable, Inject } from '@nestjs/common'
+import { Interval } from '@nestjs/schedule'
+import { InjectPinoLogger, PinoLogger } from 'nestjs-pino'
+import { catchError, firstValueFrom, map } from 'rxjs'
+import { TokensSymbols } from './dtos/array-tokens-symbol'
+import { Token, CoinMarketTokenPrice, TokenPrice } from './tokens.interface'
+
+@Injectable()
+export class TokensService {
+  constructor(
+    @InjectPinoLogger(TokensService.name)
+    @Inject(CACHE_MANAGER)
+    private readonly logger: PinoLogger,
+    private readonly httpService: HttpService,
+    private cacheManager: Cache,
+  ) {}
+
+  async getTokensPrice(symbol: string[]): Promise<Token[]> {
+    const tokens: Token[] | undefined = await this.cacheManager.get('tokens')
+    console.log(tokens)
+    return !tokens ? [] : tokens.filter((element) => symbol.includes(element.symbol))
+  }
+
+  async fetchTokensPrices(symbols: string[]): Promise<CoinMarketTokenPrice> {
+    const response = await this.httpService
+      .get('/cryptocurrency/quotes/latest', {
+        params: {
+          symbol: symbols.join(','),
+        },
+      })
+      .pipe(
+        map((res) => res.data.data),
+        catchError((error: any) => {
+          this.logger.error({ error }, 'Error fetching tokens prices')
+          throw new Error(error?.response?.data.message || error.message)
+        }),
+      )
+
+    const tokens = await firstValueFrom(response)
+    return tokens
+  }
+  //It is suggested to use 23529 if switching to the Hobbyist plan259200
+  @Interval(5000)
```

Comment:
Try to handle this value with an environment variable. For example:

const UPDATE_PRICES_FRECUENCY_MS = process.env.UPDATE_PRICES_FRECUENCY_MS

@Interval(UPDATE_PRICES_FRECUENCY_MS)

Also add this new var to env.sample withe the intervals for free and hobbyist plan

for example:

```
# 60000 Hobbyist / every minute
# 267840 free / every 4.4m
UPDATE_PRICES_FRECUENCY_MS=
```